src/Symfony/Component/Console/Application.php

```diff
@@ -282,19 +284,25 @@ public function doRun(InputInterface $input, OutputInterface $output)
         }
 
         if ($this->dispatcher) {
-            foreach ($this->signalsToDispatchEvent as $signal) {
-                $event = new ConsoleSignalEvent($command, $input, $output, $signal);
+            if (!SignalRegistry::isSupported()) {
+                if ($this->dispatcher instanceof ComponentEventDispatcherInterface && $this->dispatcher->hasListeners(ConsoleEvents::SIGNAL)) {
```

Comment:
What'as about checking that `count($this->signalsToDispatchEvent) > 0` to let people use the application without pcntl. Even if something (like a 3rd party bundle) listen to `ConsoleEvents::SIGNAL`

Reply:
👍  Updated to just not set `signalsToDispatchEvent` if pcntl is not available